operator/statuses.yaml

```diff
@@ -0,0 +1,194 @@
+# This file describes the unified cluster statuses generated for v1 and v2
+# it's used by gen status to generate consistently typed structures for our errors.
+# 
+# It should remain heavily documented to demonstrate usage.
+# 
+# General usage:
+# 
+#    status := statuses.NewMyKind()
+#    status.SetMyCondition(statues.MyKindMyConditionMyReason, "something happened!") // or don't pass any strings for reasons that have default messages
+# 
+#    // for server side applies you can do
+#    kubernetes.ApplyPatch(config.WithStatus(mypackageac.SchemaStatus()
+#      .WithConditions(status.StatusConditionConfigs(myKind)...)))
+# 
+#    // or for a standard update (internally this acts like apimeta.SetStatusCondition)
+#    if status.UpdateConditions(myKind) { client.Update(...) }
+
+# kind is the name of the status struct, here we'll generate a struct named "Cluster"
+- kind: Cluster
+  # appliesTo allows us to generate struct and field kubebuilder annotations to default our conditions in our API structs
+  appliesTo:
+    - redpanda/v1alpha2/Redpanda
```

Comment:
Not related, but the next would be to add Cluster custom resource?
```suggestion
- kind: Cluster
  # appliesTo allows us to generate struct and field kubebuilder annotations to default our conditions in our API structs
  appliesTo:
    - redpanda/v1alpha2/Redpanda
    - vectorized/v1alpha1/Cluster
```

Reply:
Yep